src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Result/ShouldRunTestResult.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// The possible outcomes of <see cref="Validator.ShouldRunAsync(ValidationContext)"/>.
+    /// </summary>
+    public enum ShouldRunTestResult
+    {
+        /// <summary>
+        /// The test should run.
+        /// </summary>
+        Yes,
+
+        /// <summary>
+        /// The test can be safely skipped.
+        /// </summary>
+        No,
+
+        /// <summary>
+        /// It could not be determined if the test should be run.
+        /// The test must be attempted again when more information is available.
```

Comment:
I would mention that the validation will be done by a future leaf and not the current leaf.

Reply:
Added your feedback to all the docs.